reconcile/utils/jinja2/filters.py

```
@@ -125,21 +125,28 @@ def matches_jsonpath(input: Any, jsonpath: str | None) -> bool:
+    for part in jsonpath_parts(path, ignore_root=True):
+        if isinstance(part, jsonpath_ng.Fields):
+            # RFC 6901: escape ~ as ~0 and / as ~1, in that order, before joining
+            segments.extend(
+                str(f).replace("~", "~0").replace("/", "~1") for f in part.fields
+            )
+        elif isinstance(part, jsonpath_ng.Index):
+            segments.extend(str(i) for i in part.indices)
+    return "/" + "/".join(segments)
 
 
 def json_pointers(input: Any, jsonpath: str) -> list[str]:
```

Ruff (0.16.0)

[error] Function argument `input` is shadowing a Python builtin

(A002)